supercollider/server.py

```diff
@@ -13,188 +16,134 @@ def __init__(self, hostname="127.0.0.1", port=57110):
-            self.osc_server = None
-        self.osc_server_thread.join()
-
-    def sync(self):
-        """
-        Causes the server to immediately execute and flush all asynchronous commands.
-        This command is always synchronous so blocks the current thread.
 
-        This can be used as a `ping` functionality to check that the server is running
-        and responding.
-        """
-        self._send_msg("/sync")
-        return self._await_response("/synced", None, lambda n: n)
-    
-    def query_tree(self, group=None):
-        self._send_msg("/g_queryTree", group.id if group else 0, 0)
-        return self._await_response("/g_queryTree.reply")
+        self.sync()
 
-    def get_status(self, callback=None, blocking=True):
-        """
-        Query the current Server status, including the number of active units, CPU
-        load, etc.
-
-        Example:
-            >>> server.status
-            {
-                'num_ugens': 5,
-                'num_synths': 1,
-                'num_groups': 2,
-                'num_synthdefs': 107,
-                'cpu_average': 0.08170516043901443,
-                'cpu_peak': 0.34912213683128357,
-                'sample_rate_nominal': 44100.0,
-                'sample_rate_actual': 44100.07866992249
-            }
-        """
-        def _handler(args):
-            args_dict = {
+    # ------------------- Handlers ------------------- #
+    # TODO: Move handlers to a separate file or class
+    def dummy_handler(self, address, *args):
+        pass
+
+    def simple_handler(self, address, *args):
+        return args
+
+    def param_handler(self, address, *args):
+        return args[2]
+
+    def buf_handler(self, address, *args):
+        #print(f"Buffer Handler - {address}: {args}")
+        return args[3:]
+
+    def status_handler(self, address, *args):
+        status_dict = {
                 "num_ugens": args[1],
                 "num_synths": args[2],
                 "num_groups": args[3],
                 "num_synthdefs": args[4],
                 "cpu_average": args[5],
                 "cpu_peak": args[6],
                 "sample_rate_nominal": args[7],
-                "sample_rate_actual": args[8],
-            }
-            if callback:
-                callback(args_dict)
-            return args_dict
+                "sample_rate_actual": args[8]}
+            
+        return status_dict
+
+    def version_handler(self, *args):
+        version_dict = {
+                "program_name": args[1],
+                "version_major": args[2],
+                "version_minor": args[3],
+                "version_patch": args[4],
+                "git_branch": args[5],
+                "commit_hash": args[6]}
+            
+        return version_dict
+    
+    # ------------------- Client Messages ------------------- #
+    def _send_msg(self, address, *args):
+        self.sc_client.send_message(address, [*args])
 
+    def sync(self, num = 1):
```

Comment:
`num`: Should be a more self-explanatory argument name - what does `num` represent?